core/trino-parser/src/main/java/io/trino/sql/tree/Label.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.tree;
+
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public final class Label
+{
+    private final String name;
+
+    public static Label from(Identifier identifier)
```

Comment:
This is not ideal. This method doesn't belong in the parser-related classes. It has semantic behavior (transforming based on canonical value) that should be part of the planning phase.

In fact, the particular form of Label returned by this method seems to be part of the IR, not the AST.